src/fstests/fstests.jinja2

```diff
@@ -135,30 +142,24 @@ class Job(BaseJob):
         return node
 
     def _run(self, src_path):
-        fail_results = {
-            'node': {
-                'name': 'fstests',
-                'result': 'fail',
-                'state': 'done',
-            },
-            'child_nodes':[],
-        }
-        if not self._kernel_config(src_path):
-            return fail_results
-        if not self._kernel_build(src_path):
-            return fail_results
-        if not self._run_smoke_tests():
-            return fail_results
-        if not self._get_xml_from_vm(src_path):
-            return fail_results
+        results = self._generate_node_result('fstests', 'fail', 'done')
+        config_status, result_node = self._kernel_config(src_path)
+        results['child_nodes'].append(result_node)
+        if not config_status:
+            return results
+        build_status, result_node = self._kernel_build(src_path)
+        results['child_nodes'].append(result_node)
+        if not build_status or not self._run_smoke_tests() or not self._get_xml_from_vm(src_path):
+            return results
         try:
-            results = self._parse_results(src_path)
+            parsed_results = self._parse_results(src_path)
+            results['child_nodes'].extend(parsed_results['child_nodes'])
             results['node']['result'] = 'pass'
             results['node']['state'] = 'done'
```

Comment:
The node `state` is already been set to `done` in `results = self._generate_node_result('fstests', 'fail', 'done')`
This line can be dropped.
```suggestion

```